core/lib/web3_decl/src/error.rs

```diff
@@ -42,3 +43,65 @@ pub enum Web3Error {
     #[error("Tree API is not available")]
     TreeApiUnavailable,
 }
+
+use std::{collections::HashMap, error, fmt, fmt::Debug};
+
+#[derive(Debug)]
+pub struct RpcErrorWithDetails {
+    pub inner_error: RpcError,
```

Comment:
It looks weird for errors to have public fields. I'd suggest to make fields private and define a getter for `inner_error` (e.g., `fn inner(&self) -> &RpcError`) instead.